File: read_cap_weight.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-

"""
Script to read information from weight files in cap
"""

def read_cap_weight(filename):
    f = open(filename, "r")
    lines = f.readlines()
    f.close()
    count = 0
    stnm = [];edist = [];PV_wt = [];PR_wt = [];SV_wt = [];SR_wt = [];ST_wt = []; P_arrival = []; P_len =[]; S_arrival = []; S_len = []; waveform_shft = [];

    for line in lines:
        line_elements = line.split()
        stnm.append(line_elements[0])
        edist.append(float(line_elements[1]))
        PV_wt.append(float(line_elements[2]))
        PR_wt.append(float(line_elements[3]))
        SV_wt.append(float(line_elements[4]))
        SR_wt.append(float(line_elements[5]))
        ST_wt.append(float(line_elements[6]))
        P_arrival.append(float(line_elements[7]))
        P_len.append(float(line_elements[8]))
        S_arrival.append(float(line_elements[9]))
        S_len.append(float(line_elements[10]))
        waveform_shft.append(float(line_elements[11]))
        
    return stnm,edist,PV_wt,PR_wt,SV_wt,SR_wt,ST_wt,P_arrival,P_len,S_arrival,S_len,waveform_shft
    

File: UTILS/matlab/read_cap_weight.m
function [stnm, pol, edist, PV_wt, PR_wt, SV_wt, SR_wt, ST_wt, P_arrival, P_len, S_arrival, S_len, waveform_shft] = read_cap_weight(weight_filename)

% get number of columns
% THIS STEP OUGHT TO BE UNNECESSARY IF WE CAN USE fscanf OR textscan PROPERLY
[~,ncol] = system(sprintf("head -1 %s | tr '|' ' ' | wc -w",weight_filename));
ncol = str2num(ncol);

fid = fopen(weight_filename);
if ncol==12
    C = textscan(fid,'%s %f %f %f %f %f %f %f %f %f %f %f');
elseif ncol==13
    C = textscan(fid,'%s %f %f %f %f %f %f %f %f %f %f %f %f');
else
    ncol
    weight_filename
    error('can only have 12 or 13 columns in cap weight file');
end
fclose(fid);

stnm_pol = C{1};
% Get polarities
for ii = 1:length(stnm_pol)
    tmp = strsplit(stnm_pol{ii},'/');
    stnm{ii} = tmp{1};
    if length(tmp)==2
        pol(ii) = str2double(tmp{2});
    else
        pol(ii) = 0;
    end
end
edist = C{2};
PV_wt = C{3};
PR_wt = C{4};
SV_wt = C{5};
SR_wt = C{6};
ST_wt = C{7};
P_arrival = C{8};
P_len = C{9};
S_arrival = C{10};
S_len = C{11};
waveform_shft = C{12};

% EXamples
if 0
    weight_filename = '/home/vipul/REPOSITORIES/GEOTOOLS/python_util/util_data_syn/20090407201253480/weight.dat';
    [stnm, pol, edist, PV_wt, PR_wt, SV_wt, SR_wt, P_arrival, P_len, S_arrival, S_len, waveform_shft] = read_cap_weight(weight_filename);
    weight_filename = '/home/vipul/REPOSITORIES/GEOTOOLS/python_util/util_data_syn/20090407201253480/weight_test.dat';
    write_cap_weight(weight_filename, stnm, pol, edist, PV_wt,PR_wt, SV_wt, SR_wt, ST_wt, P_arrival, P_len, S_arrival, S_len, waveform_shft)
end
